src/txn/txn_timestamp.c

```diff
@@ -41,7 +41,8 @@ __wt_timestamp_to_hex_string(wt_timestamp_t ts, char *hex_timestamp)
 		return;
 	}
 	if (ts == WT_TS_MAX) {
-		(void)strcpy(hex_timestamp, "ffffffffffffffff");
+		(void)strncpy(
```

Comment:
Clang Tidy doesn't like `strcpy`. It suggests `strlcpy` but that appears to be a BSD thing so I've gone for `strncpy` instead and it seems happy with that.

Reply:
Static analyzers tend not to like `strncpy` because it doesn't always nul-terminate. Maybe `memcpy`?

```
--- txn_timestamp.c     2019-03-04 10:45:49.398128000 -0500
+++ -   2019-03-04 10:56:11.712206653 -0500
@@ -41,8 +41,9 @@
                return;
        }
        if (ts == WT_TS_MAX) {
-               (void)strncpy(
-                   hex_timestamp, "ffffffffffffffff", WT_TS_HEX_STRING_SIZE);
+#define        WT_TS_MAX_HEX_STRING    "ffffffffffffffff"
+               (void)memcpy(hex_timestamp,
+                   WT_TS_MAX_HEX_STRING, strlen(WT_TS_MAX_HEX_STRING) + 1));
                return;
        }
```
I don't feel strongly at all, just tossing that idea out if you like it.

Reply:
faca49d
This seems like a better idea. I've switched over to `memcpy`.